CSharp_Course_Project/Task4_ATMMachines.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+
+namespace CSharp_Course_Project
+{
+    class Task4_ATMMachines
+    {
+        public static void ValidatePin()
+        {
+            while (true)
+            {
+                Console.WriteLine("Enter the 4-digit or 6-digit PIN:");
+                string pin = Task4_ATMMachines.InputPin();
+                bool isPinNumber = Task4_ATMMachines.IsPinNumber(pin);
+                bool correctPinLength = Task4_ATMMachines.CorrectPinLength(pin);
+                if (!isPinNumber)
+                {
+                    Console.WriteLine($"PIN is not a number: {pin}. Try again...");
+                    continue;
+                }
+                else if (!correctPinLength)
+                {
+                    Console.WriteLine($"Not valid PIN length: {pin}. Try again...");
+                    continue;
+                }
+                else if (isPinNumber || correctPinLength)
```

Comment:
Incorrect logic.
The pin is correct when it consists of numbers AND has a correct length.

Reply:
fixed